hummingbot/connector/exchange/zebpay/zebpay_api_order_book_data_source.py

```diff
@@ -257,52 +260,21 @@ async def listen_for_trades(self, ev_loop: asyncio.BaseEventLoop, output: asynci
-                        # TODO Brian: Review subscription request format for Zebpay
-                        "method": "subscribe",
-                        "markets": trading_pairs,
-                        # Working assumption that Zebpay WS trades are in the "history" subscription. Require format
-                        "subscriptions": ["history"]
-                    }
-                    await ws.send(ujson.dumps(subscription_request))
-                    async for raw_msg in self._inner_messages(ws):
-                        msg = ujson.loads(raw_msg)
-                        msg_type: str = msg.get("type", None)
-                        if DEBUG:
-                            self.logger().debug(f'<<<<< ws msg: {msg}')
-                        if msg_type is None:
-                            raise ValueError(f"Zebpay Websocket message does not contain a type - {msg}")
-                        elif msg_type == "error":
-                            raise ValueError(f"Zebpay Websocket received error message - {msg['data']['message']}")
-                        elif msg_type == "trades":
-                            # TODO Brian: Confirm lastModifiedDate key parsing is correct
-                            trade_timestamp: float = pd.Timestamp(msg["lastModifiedDate"], unit="ms").timestamp()
-                            trade_msg: OrderBookMessage = ZebpayOrderBook.trade_message_from_exchange(msg,
-                                                                                                      trade_timestamp)
-                            output.put_nowait(trade_msg)
-                        elif msg_type == "subscriptions":
-                            self.logger().info("subscription to trade received")
-                        else:
-                            raise ValueError(f"Unrecognized Zebpay WebSocket message received - {msg}")
-                        await asyncio.sleep(0)
-            except asyncio.CancelledError:
-                raise
-            except Exception:
-                self.logger().network(
-                    f'{"Unexpected error with WebSocket connection."}',
-                    exc_info=True,
-                    app_warning_msg=f'{"Unexpected error with Websocket connection. Retrying in 30 seconds..."}'
-                                    f'{"Check network connection."}'
-                )
-                await asyncio.sleep(30.0)
+        zebpay_ws_feed = get_zebpay_ws_feed()
+        if DEBUG:
+            self.logger().info(f"ZOB.listen_for_trades new connection to ws: {zebpay_ws_feed}")
+        if self._sio is None:
+            self._sio = socketio.AsyncClient()
+            zebpayNamespace = ZebpayCustomNamespace()
+            zebpayNamespace.set_trade_queue(output)
+            self._sio.register_namespace(zebpayNamespace)
+            await self._sio.connect(zebpay_ws_feed, transports=['websocket'])
+        else:
+            zebpayNamespace = self._sio.namespace_handlers["/"]
```

Comment:
Not entirely sure this is correct. I don't see any way this execution path can occur.
Also it seems to me this should be inside a loop, with retries in case of connection errors, and exit in case of Cancelled error, like in the previous code...